services/frontend/src/components/SearchBar.js

```diff
@@ -1,18 +1,61 @@
-const SearchBar = () => {
+import { useState } from "react";
+import { PropTypes } from "prop-types";
+
+const SearchBar = ({ handleUserSearch }) => {
+  const [userInput, setUserInput] = useState("");
+
+  const handleUserInputChange = (event) => {
+    setUserInput(event.target.value);
+  };
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    try {
+      const parsedLocation = validateFormData(userInput);
+      handleUserSearch(parsedLocation);
+    } catch (e) {
+      alert(
+        "Search input could not be parsed. Please use `City, State` format."
+      );
+    }
+    // Reset input
+    setUserInput("");
+  };
+
+  const validateFormData = (stringData) => {
+    // This function will be replaced by backend parsing
+    let splitString = stringData.split(",");
+    console.log(splitString);
+    if (splitString.length !== 2) {
+      throw new Error(`Unable to parse input ${stringData}`);
+    }
+    const location = {
+      city: splitString[0].trim(),
+      state: splitString[1].trim(),
+    };
+    return location;
+  };
```

Comment:
I already have some fuzzy-matching implemented for city, state.  I'll find that

Reply:
Nice! Yeah like the comment says this should probably live in the backend world.